backend/src/test/java/trkpo/spbstu/hospitalavailability/dto/ClientRequestDtoTest.java

```diff
@@ -0,0 +1,65 @@
+package trkpo.spbstu.hospitalavailability.dto;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ClientRequestDtoTest {
+    private Validator validator;
+
+    @Before
+    public void setUp() {
+        this.validator = Validation.buildDefaultValidatorFactory().getValidator();
```

Comment:
Тут фабрику закрывать надо, она AutoClosable. Но чтобы не корячить try тут вокруг, можно через ломбок аннотацию решить вопрос (var опционален, просто так быстрее написать было, можно и норм тип))

```suggestion
        @Cleanup
        var factory = Validation.buildDefaultValidatorFactory();
        this.validator = factory.getValidator();
```

Reply:
Совсем в идеале конечно ее один раз создать и реюзать, а не дергать на каждый тест, но хотя бы вот так

Reply:
так Before же один раз перед тестами вызывает setUp()

Reply:
Не, `Methods annotated with the @Before annotation are run before each test.`

Ну и все равно тогда надо было бы чистить, даже если бы он один раз гонялся, close у фабрики он все равно не вызовет (а там в разных реализациях закрываются и очищаются бин менеджеры и тд)

Reply:
Чтобы не путаться, лучше ставить явно BeforeAll или BeforeEach аннотации. Но BeforeAll только на статический метод вешается. А вот BeforeEach - это просто более подробный вариант того, что у тебя